irony.el

```diff
@@ -661,6 +661,17 @@ If no such file exists on the filesystem the special file '-' is
                              (format "%s\n"
                                      (combine-and-quote-strings argv)))))))
 
+(defun irony--send-request-sync (request callback &rest args)
```

Comment:
Yeah that was what I was thinking about.

I was also thinking about that this way is a little bit more straightforward, and less risky if there are some other factors that I haven't foreseen.